aten/src/ATen/native/BlasKernel.cpp

```diff
@@ -196,17 +198,24 @@ void gemv(char trans, int64_t m, int64_t n, scalar_t alpha, scalar_t *a, int64_t
   } else {
     if (beta != scalar_t(1) && beta != scalar_t(0)) scal<scalar_t>(m, beta, y, incy);
 
+    std::vector<opmath_t> sum(m);
+    for (const auto i : c10::irange(m)) {
+      sum[i] = y[i * incy];
+    }
```

Comment:
You only need to use a buffer for BFloat16 and Half.

Reply:
Sorry, I did not catch you up here.
From my understanding, only use a buffer will also have inefficient memory ordering like  in `gemm_core`, are you suggesting specializing `gemv` for BF16 with inefficient memory ordering and keep efficient memory ordering for other types. (I do not want to introduce Half in this PR, and there will be another PR to enable Half with this)

Reply:
You don't need to change the iteration order, just accumulate in the result when `opmath_t` is the same as `scalar_t` (i.e. not BFloat16 and Half). 

>  do not want to introduce Half in this PR, and there will be another PR to enable Half with this

I don't understand your point here, this PR does enable opmath for Half...

Reply:
Got it. I misunderstand "A" buffer before, I thought it means use a scalar buffer.
>   I don't understand your point here, this PR does enable opmath for Half...
I just saw some comments is mentioning `Half`. We can forget `Half` it is this PR, this should be another PR for `Half`